tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+		$method->setAccessible( true );
+
+		$indexer = new Opus_SolrSearch_Index_Indexer();
+		$method->invokeArgs( $indexer, array( new DomDocument(), null, 1 ) );
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2240
+	 */
+	public function testIndexDocumentWithMultipleTitleMainInSameLanguage() {
+		$doc = new Opus_Document();
+		$doc->setServerState( 'published' );
+		$doc->setLanguage( 'eng' );
+
+		$title = new Opus_Title();
+		$title->setValue( 'foo' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleMain( $title );
+
+		$title = new Opus_Title();
+		$title->setValue( 'bar' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleMain( $title );
+
+		$doc->store();
+
+		$exception = null;
+		try {
+			$this->indexer->addDocumentsToIndex( $doc );
+		}
+		catch ( Exception $e ) {
+			$exception = $e;
+		}
+
+		$this->assertNotNull( $exception );
+		$this->assertInstanceOf( 'Opus_Search_InvalidQueryException', $exception );
+
+		$doc->deletePermanent();
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2240
+	 */
+	public function testIndexDocumentWithMultipleAbstractsInSameLanguage() {
+		$doc = new Opus_Document();
+		$doc->setServerState( 'published' );
+		$doc->setLanguage( 'eng' );
+
+		$title = new Opus_Title();
+		$title->setValue( 'foo' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleAbstract( $title );
+
+		$title = new Opus_Title();
+		$title->setValue( 'bar' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleAbstract( $title );
+
+		$doc->store();
+
+		$exception = null;
+		try {
+			$this->indexer->addDocumentsToIndex( $doc );
+		}
+		catch ( Exception $e ) {
+			$exception = $e;
+		}
+
+		$this->assertNotNull( $exception );
+		$this->assertInstanceOf( 'Opus_Search_InvalidQueryException', $exception );
+
+		$doc->deletePermanent();
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2240
+	 *
+	 * @expectedException Opus_SolrSearch_Index_Exception
+	 */
+	public function testIndexDocumentWithUnknownIndexField() {
+		$this->markTestSkipped( 'not supported in Opus_Search API' );
+
+		$xml = new DOMDocument();
+		$xml->loadXML(
+			'<add>
+                  <doc>
+                    <field name="id">987654321</field>
+                    <field name="year"/>
+                    <field name="language">de</field>
+                    <field name="author_sort"/>
+                    <field name="has_fulltext">false</field>
+                    <field name="doctype">article</field>
+                    <field name="belongs_to_bibliography">false</field>
+                    <field name="xyz_unknown_field">foo</field>
+                  </doc>
+                </add>' );
+
+		$class  = new ReflectionClass( 'Opus_SolrSearch_Index_Indexer' );
+		$method = $class->getMethod( 'sendSolrXmlToServer' );
+		$method->setAccessible( true );
+
+		$method->invoke( $this->indexer, $xml );
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2417
+	 */
+	public function testFulltextVisibilityIsConsideredInFacetForFrontdoorVisibleFulltext() {
```

Comment:
As far as I understand this it is related to assessing use of invalid XSLT for transforming Opus XML into (malformed) Solr XML has to be detected somehow. However, what code is expected to detect this? Is it Solr rejecting file due to mismatching schema? I think it does so. Is it Solarium? Since it's brokering, only, I guess this isn't the place to detect it. Is it Opus? Since Solr has all the knowledge on validating I'd suspect testing this in Opus would make things very redundant.

So, finally this test is about badly transforming. Thus I'd expect to run test with a explicitly invalid transformation file trying to index document expecting Solr to reject it with an error as generic as documented here. XSLT is selected per configuration and thus might be switched just like in any other test.